modules/standard/BigInteger.chpl

```diff
@@ -2767,88 +2825,95 @@ module BigInteger {
       return isPrime;
   }
 
-  proc bigint.nextprime(const ref a: bigint) {
+  proc nextprime(ref result: bigint, const ref a: bigint) {
     if _local {
-      mpz_nextprime(this.mpz, a.mpz);
-
-    } else if this.localeId == chpl_nodeID &&
-              a.localeId    == chpl_nodeID {
-      mpz_nextprime(this.mpz, a.mpz);
-
+      mpz_nextprime(result.mpz, a.mpz);
+    } else if result.localeId == chpl_nodeID &&
+              a.localeId == chpl_nodeID {
+      mpz_nextprime(result.mpz, a.mpz);
```

Comment:
Similar comment as below, let's change this to just check the result locale and then localize `a` so we take the `else` less.